File: i150013_Sabkat/q6.py
#q6
def count( S, m, n ): 
    table=[[0 for x in range(m+1)] for x in range(n+1)] 
    for i in range(1,m):
        table[0][i] = 1; 
 

    for i in range(1,m): 
        for j in range(1,n): 
            if(S[i-1]>j): 
                table[i][j]=table[i-1][j] 
            else:
                 table[i][j]=table[i-1][j]+table[i][j-(i-1)]
    for i in range(1,m): 
        for j in range(1,n):
            print(table[i][j],end=" ")
        print("\n")
    return table[m-1][n-1] 
 
arr = [1, 2, 3] 
m = len(arr) 
N = 4
x = count(arr, m, N) 
print (x) 

File: i141638_UmerHasnain/Q6.cpp
#include <iostream>
using namespace std;
int count(int S[], int m, int n)
{
	if (n == 0) 
        return 1; 
        
    if (n < 0) 
        return 0; 
    
    if (m <=0 && n >= 1) 
        return 0; 
        
    return count( S, m - 1, n ) + count( S, m, n-S[m-1] ); 
}

int main()
{
	int i, j; 
    int arr[] = {1, 3, 5,7}; 
    int m = sizeof(arr)/sizeof(arr[0]); 
    cout<<"Total No. of Ways: "<<count(arr, m, 9)<<endl;
    return 0; 
}
